### ISSUE TITLE
How to make the final model smaller after training interruption？

### ISSUE THREAD

**Issue by 992060356** [ultralytics/yolov5#3883]
## ❔Question
I use yolov5l6 pre training model to train my own data set, a total of 300 epochs. At the 121st epoch, the map is close to convergence, but the final best.pt and last.pt are 365MB. I remember when I trained to the final epoch, what method would the final model use to compress and shrink. If I interrupt the training like this, how can I make the model smaller?

**Comment by wudashuo** [comment]
You can use `strip_optimizer` in `utils/general.py` to get the model smaller.

**Comment by glenn-jocher** [comment]
@992060356 yes you can run `strip_optimizer('path/to/model.pt')` on any model you want manually as @wudashuo explained.

YOLOv5 🚀 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

**Comment by 992060356** [comment]
> You can use `strip_optimizer` in `utils/general.py` to get the model smaller.

Thank you very much for your reply, I will have a try.

**Comment by 992060356** [comment]
> @992060356 yes you can run `strip_optimizer('path/to/model.pt')` on any model you want manually as @wudashuo explained.
> 
> YOLOv5 🚀 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
> https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439
> 
> The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
> https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509
> 
> Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
> 
> <img alt="YOLOv5 Models" width="700" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png">

Thank you,you can answer my erery question its really nice!
